### PR TITLE
Common math types (vec, quat, mat, etc) and LocalTransform, implemented by Realtime Math

### PR DESCRIPTION
**Pull request by MarshallW906**

### CODE REVIEW

**Review by Top61ly — APPROVED** [review]
好狠啊，这dx

**Comment by shurunxuan** [comment]
我等SimpleMath干掉之后approve

**Comment by MarshallW906** [comment]
macOS Clang will always fail to compile because the `macos:latest` on GitHub Actions is macOS 12, with Apple Clang 14, which does not support `std::bit_cast`

**Review by Top61ly — APPROVED** [review]
Big YES!

File: sample/App/SampleApp.cpp [hunk 1]
```diff
@@ -32,6 +33,7 @@ void SampleApp::Initialize()
     gameObject->SetName("TestObject");
 
     TestComponent* testComponent = gameObject->AddComponent<TestComponent>();
+    gore::Transform* transform   = gameObject->AddComponent<gore::Transform>();
```
Comment (shurunxuan) [inline sample/App/SampleApp.cpp]: How about making Transform a default component (as in, automatically add it to new objects)?

**Review by shurunxuan — APPROVED** [review]
lgtm

File: sample/App/SampleApp.cpp (hunk 1 above)
Comment (MarshallW906) [reply sample/App/SampleApp.cpp]: Ah you are right. I forgot to do this after a long journey with math libraries :w 
Wait for me

**Pull request merged**